R/valid.R

```diff
@@ -84,11 +84,21 @@ validate <- function(object, recursive = TRUE, properties = TRUE) {
 
   # Next, recursively validate the object
   errors <- character()
-  repeat({
-    errors <- c(errors, class_validate(class, object))
+  repeat {
+    error <- class_validate(class, object)
```

Comment:
Is there enough duplication here it's worth pulling out into a function?

Reply:
The usage patterns between prop validators and class validators are subtly different, but different enough that I believe a "generalized" function wouldn't be worth the added complexity.